src/assets/scripts/client/airline/AirlineModel.js

```diff
@@ -68,26 +68,15 @@ export default class AirlineModel extends BaseModel {
          */
         this.flightNumberGeneration = {
```

Comment:
can you add a `TODO` here about abstracting the flightNumberGenerator out to it's own helper.  There is a growing amount of logic here and I feel it will likely (though maybe not now) need to be abstracted.  

The `AirlineModel` shouldn't really care about the _how_ only know about the rules.  It should be something like: _Hey, given these rules, build me a flight number'_

Reply:
Its acutually called the callsign, not the flight number :D

Reply:
I believe we're using it for both here

Reply:
I think (depending on the airline), they are usually different, for example, the flight number never contains letters, the callsign can.

Reply:
@willfrd we don't display the flight number here, (ie FR6251) so it really doesn't matter what it's called

Reply:
Yeahhh, just saying :-)